tasmota/xdrv_04_light.ino

```diff
@@ -1257,6 +1257,28 @@ void LightSetBri(uint8_t device, uint8_t bri) {
   }
 }
 
+void LightSetBriScaled(uint8_t bri) {
+  // change both dimmers, retain ratio between white and color channels
+  uint32_t bri_rgb = light_state.getBriRGB();
+  uint32_t bri_ct = light_state.getBriCT();
+#ifdef DEBUG_LIGHT
+  AddLog(LOG_LEVEL_DEBUG, "LightSetBri bri:%d, bri_rgb:%d, bri_ct: %d", bri, bri_rgb, bri_ct);
+#endif
+  uint32_t max_bri = light_state.getBri();
+  if (max_bri == 0) {
+    bri_rgb = bri;
+    bri_ct = bri;
+  } else {
+    bri_rgb = bri * bri_rgb / max_bri;
```

Comment:
prefer: it will handle better rounding
`bri_rgb = changeUIntScale(brI_rgb, 0, max_bri, 0, bri)`
`bri_ct = changeUIntScale(bri_ct, 0, max_bri, 0, bri)`